Stack_in_C_dynamic

```diff
@@ -0,0 +1,63 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+
+
+
+typedef enum
+{
+    STACK_CHAR,
+    STACK_INT
+
+}datatype;
+
+
+
+typedef struct stack
+{
+    int type_size;
+    size_t size;
+    void* data;
+    void* top;
+}stack;
+
+stack stackCreator(size_t size, datatype type){
+
+    /* Check which datatype we store in order to allocate memory for it*/
+    int effectiveSize;
+    if (type == STACK_CHAR){
+        effectiveSize = sizeof(char);
+    } else {
+        effectiveSize = sizeof(int);
+    }
+
+    /* Creating a stack*/
+    stack s = {
+        .size = size,
+        .type_size = effectiveSize,
+        .data = malloc(effectiveSize),
+        .top = NULL
+
+    };
+
+    return s;
+    
+
+
+}
+bool isfull(stack *s) {
+  if (s->top == (s->size*s->type_size)+s->data)
+    return true;
+  else
+    return false;
+}
+void push(stack* s,void *item){
+
+    s->top++;
+    s->data[s->top] = item;
```

Comment:
this is will not work.

Reply:
check top if NULL. and if not do memset for item to top(think about what should be the size of the memory you want to set).